switchboard-function/src/main.rs

```diff
@@ -0,0 +1,132 @@
+    ixn_data.append(&mut prize_result_bytes);
+
+    // ACCOUNTS:
+    // 1. Competition (mut)
+    // 2. Switchboard Function
+    // 3. Switchboard Function Request
+    // 4. Enclave Signer (signer): our Gramine generated keypair
+    let receive_randomness_ixn = Instruction {
+        program_id: params.program_id,
+        data: ixn_data,
+        accounts: vec![
+            AccountMeta::new(params.competition_key, false),
+            AccountMeta::new_readonly(runner.function, false),
+            AccountMeta::new_readonly(runner.function_request_key.unwrap(), false),
+            AccountMeta::new_readonly(runner.signer, true),
+        ],
+    };
+
+    // Then, write your own Rust logic and build a Vec of instructions.
+    // Should  be under 700 bytes after serialization
+    let ixs: Vec<solana_program::instruction::Instruction> = vec![receive_randomness_ixn];
+
+    // Finally, emit the signed quote and partially signed transaction to the functionRunner oracle
+    // The functionRunner oracle will use the last outputted word to stdout as the serialized result. This is what gets executed on-chain.
+    runner.emit(ixs).await.unwrap();
+}
+
+fn generate_randomness(min: u128, max: u128) -> u128 {
+    if min == max {
+        return min;
+    }
+    if min > max {
+        return generate_randomness(max, min);
+    }
+
+    // We add one so its inclusive [min, max]
+    let window = (max + 1) - min;
+
+    let mut bytes: [u8; 16] = [0u8; 16];
+    Gramine::read_rand(&mut bytes).expect("gramine failed to generate randomness");
+    let raw_result: &[u128] = bytemuck::cast_slice(&bytes[..]);
+
+    (raw_result[0] % window) + min
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // 1. Check when lower_bound is greater than upper_bound
+    #[test]
+    fn test_generate_randomness_with_flipped_bounds() {
+        let min = 100;
+        let max = 50;
+
+        let result = generate_randomness(100, 50);
+        assert!(result >= max && result < min);
+    }
+
+    // 2. Check when lower_bound is equal to upper_bound
+    #[test]
+    fn test_generate_randomness_with_equal_bounds() {
+        let bound = 100;
+        assert_eq!(generate_randomness(bound, bound), bound);
+    }
+
+    // 3. Test within a range
+    #[test]
+    fn test_generate_randomness_within_bounds() {
+        let min = 100;
+        let max = 200;
+
+        let result = generate_randomness(min, max);
+
+        assert!(result >= min && result < max);
+    }
+
+    // 4. Test randomness distribution (not truly deterministic, but a sanity check)
+    #[test]
+    fn test_generate_randomness_distribution() {
+        let min = 0;
+        let max = 9;
+
+        let mut counts = vec![0; 10];
+        for _ in 0..1000 {
+            let result = generate_randomness(min, max);
+            let index: usize = result as usize;
+            counts[index] += 1;
+        }
+
+        // Ensure all counts are non-zero (probabilistically should be the case)
```

Comment:
why not just keep pulling until they all have one (to avoid random failures)

Reply:
i just copied this lol